SQRLDotNetClientUI/ViewModels/ChangePasswordViewModel.cs

```diff
@@ -0,0 +1,240 @@
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private void PaswordStrengthScoreUpdated(object sender, ScoreUpdatedEventArgs e)
+        {
+            Dispatcher.UIThread.Post(() =>
+            {
+                string ratingText = _loc.GetLocalizationValue("PasswordStrength") + ": ";
+
+                switch (e.Score.Rating)
+                {
+                    case PasswordRating.POOR:
+                        ratingText += _loc.GetLocalizationValue("PasswordStrengthRatingPoor");
+                        this.PasswordRatingColor = BRUSH_POOR;
+                        break;
+
+                    case PasswordRating.MEDIUM:
+                        ratingText += _loc.GetLocalizationValue("PasswordStrengthRatingMedium");
+                        this.PasswordRatingColor = BRUSH_MEDIUM ;
+                        break;
+
+                    case PasswordRating.GOOD:
+                        ratingText += _loc.GetLocalizationValue("PasswordStrengthRatingGood");
+                        this.PasswordRatingColor = BRUSH_GOOD;
+                        break;
+                }
+
+                this.UppercaseIndicatorColor = e.Score.UppercaseUsed ? BRUSH_GOOD : BRUSH_POOR;
+                this.LowercaseIndicatorColor = e.Score.LowercaseUsed ? BRUSH_GOOD : BRUSH_POOR;
+                this.DigitsIndicatorColor = e.Score.DigitsUsed ? BRUSH_GOOD : BRUSH_POOR;
+                this.SymbolsIndicatorColor = e.Score.SymbolsUsed ? BRUSH_GOOD : BRUSH_POOR;
+
+                this.PasswordStrengthRating = ratingText;
+                this.PasswordStrength = (double)e.Score.StrengthPoints;
+            });
+
+            
+        }
+
+        /// <summary>
+        /// Checks if the password verification matches the new password
+        /// and enables/disables UI controls accordingly.
+        /// </summary>
+        private void CheckPasswordVerification()
+        {
+            if (!string.IsNullOrEmpty(this.NewPassword) && this.NewPassword == this.NewPasswordVerify)
+            {
+                this.CanSave = true;
+            }
+            else this.CanSave = false;
+        }
+
+        /// <summary>
+        /// Closes the current view and displays the main screen.
+        /// </summary>
+        public void Close()
+        {
+            ((MainWindowViewModel)_mainWindow.DataContext).Content =
+                ((MainWindowViewModel)_mainWindow.DataContext).MainMenu;
+        }
+
+        /// <summary>
+        /// Decrypts the current identity's block 1 using the old password
+        /// and tries to re-encrypt it using the new password.
+        /// </summary>
+        public async void SetNewPassword()
+        {
+            CanSave = false;
+
+            var progress = new Progress<KeyValuePair<int, string>>();
+            var progressDialog = new ProgressDialog(progress);
+            progressDialog.HideFinishedItems = false;
+            progressDialog.ShowDialog(_mainWindow);
```

Comment:
It's complaining about lack of await here... I'm guessing we don't want that since we want the actual decrypting to take place... do we need to suppress these?
Or call await, I think it's smart enough to not stop as long as the awaited result isn't used anywhere.

Reply:
Exactly right, we don't want to block here, since we've got stuff to do while the `ProgressDialog` is showing. I'll look into suppressing the warning, though, so thanks for pointing it out.